src/rgw/rgw_amqp.cc

```diff
@@ -883,11 +886,13 @@ class Manager {
     }
     // if error occurred during creation the creation will be retried in the main thread
     ++connection_count;
-    auto conn = connections.emplace(tmp_id, std::make_unique<connection_t>(cct, info, verify_ssl, ca_location)).first->second.get();
-    ldout(cct, 10) << "AMQP connect: new connection is created. Total connections: " << connection_count << dendl;
-    if (!new_state(conn, tmp_id)) {
-      ldout(cct, 1) << "AMQP connect: new connection '" << to_string(tmp_id) << "' is created. but state creation failed (will retry). error: " <<
-        status_to_string(conn->status) << " (" << conn->reply_code << ")"  << dendl;
+    auto new_conn = std::make_unique<connection_t>(cct, info, verify_ssl, ca_location);
+    if (new_state(new_conn.get(), tmp_id)) {
+        auto conn = connections.emplace(tmp_id, std::move(new_conn)).first->second.get();
+        ldout(cct, 10) << "AMQP connect: new connection is created. Total connections: " << connection_count << dendl;
+    } else {
+        ldout(cct, 1) << "AMQP connect: new connection '" << to_string(tmp_id) << "' is created. but state creation failed (will retry). error: " <<
+            status_to_string(conn->status) << " (" << conn->reply_code << ")"  << dendl;
```

Comment:
how does it compile, shouldn't we use "new_conn" here?

Reply:
Yes, same as above.